cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/DelayedHttpServiceResponder.java

```diff
@@ -58,109 +62,74 @@ public DelayedHttpServiceResponder(HttpResponder responder, MetricsContext metri
     this.metricsContext = metricsContext;
   }
 
-  /**
-   * Sends JSON response back to the client with status code 200 OK.
-   *
-   * @param object the object that will be serialized into JSON and sent back as content
-   */
   @Override
   public void sendJson(Object object) {
     sendJson(HttpResponseStatus.OK.getCode(), object);
   }
 
-  /**
-   * Sends JSON response back to the client.
-   *
-   * @param status the status of the HTTP response
-   * @param object the object that will be serialized into JSON and sent back as content
-   */
   @Override
   public void sendJson(int status, Object object) {
     sendJson(status, object, object.getClass(), GSON);
   }
 
-   /**
-    * Sends JSON response back to the client using the given {@link Gson} object.
-    *
-    * @param status the status of the HTTP response
-    * @param object the object that will be serialized into JSON and sent back as content
-    * @param type the type of object
-    * @param gson the Gson object for serialization
-   */
   @Override
   public void sendJson(int status, Object object, Type type, Gson gson) {
-    send(status, Charsets.UTF_8.encode(gson.toJson(object, type)), "application/json", null, false);
+    doSend(status, Charsets.UTF_8.encode(gson.toJson(object, type)), "application/json", null, false);
   }
 
-  /**
-   * Sends a UTF-8 encoded string response back to the HTTP client with a default response status.
-   *
-   * @param data the data to be sent back
-   */
   @Override
   public void sendString(String data) {
     sendString(HttpResponseStatus.OK.getCode(), data, Charsets.UTF_8);
   }
 
-  /**
-   * Sends a string response back to the HTTP client.
-   *
-   * @param status the status of the HTTP response
-   * @param data the data to be sent back
-   * @param charset the Charset used to encode the string
-   */
   @Override
   public void sendString(int status, String data, Charset charset) {
-    send(status, charset.encode(data), "text/plain; charset=" + charset.name(), null, false);
+    doSend(status, charset.encode(data), "text/plain; charset=" + charset.name(), null, false);
```

Comment:
Either is fine. It's a private method and the existing logic already handle null.